src/theory/bags/inference_generator.h

```diff
@@ -367,6 +367,139 @@ class InferenceGenerator
    */
   Node getMultiplicityTerm(Node element, Node bag);
 
+  /**
+   * @param n has form ((_ table.group n1 ... nk) A) where A has type T
+   * @return an inference that represents:
+   * (=>
+   *  (= A (as bag.empty T))
+   *  (= skolem (bag (as bag.empty T) 1))
+   * )
+   * where skolem is a variable equals ((_ table.group n1 ... nk) A)
+   */
+  InferInfo groupNotEmpty(Node n);
+  /**
+   * @param n has form ((_ table.group n1 ... nk) A) where A has type (Table T)
+   * @param e an element of type T
+   * @param part a function of type T -> (Table T)
+   * @return an inference that represents:
+   * (=>
+   *   (bag.member x A)
+   *   (and
+   *     (= (bag.count (part x) skolem) 1)
+   *     (= (bag.count x (part x)) (bag.count x A))
+   *     (= (bag.count (as bag.empty (Table T)) skolem) 0)
+   *   )
+   * )
+   *
+   * where skolem is a variable equals ((_ table.group n1 ... nk) A)
+   */
+  InferInfo groupUp1(Node n, Node x, Node part);
+  /**
+   * @param n has form ((_ table.group n1 ... nk) A) where A has type (Table T)
+   * @param e an element of type T
+   * @param part a function of type T -> (Table T)
+   * @return an inference that represents:
+   * (=>
+   *   (= (bag.count x A) 0)
+   *   (= (part x) (as bag.empty (Table T)))
+   * )
+   * where skolem is a variable equals ((_ table.group n1 ... nk) A)
+   */
+  InferInfo groupUp2(Node n, Node x, Node part);
+  /**
+   * @param n has form ((_ table.group n1 ... nk) A) where A has type (Table T)
+   * @param B an element of type (Table T)
+   * @param x an element of type T
+   * @return an inference that represents:
```

Comment:
Missing `part`.

Reply:
Added to this function and other functions as well.